File: python/length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        j = 0
        start = False
        for i in reversed(range(0, len(s))):
            if s[i] != " ":
                start = True
                j += 1
            elif start and s[i] == " ":
                break
        return j

File: rust/length-of-last-word.rs
impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        return s
            .split(" ")
            .into_iter()
            .filter(|w| !w.is_empty())
            .last()
            .unwrap()
            .len() as i32;
    }
}

impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        let mut i = 0;
        let mut start = false;
        for (index, c) in s.chars().rev().enumerate() {
            if c != ' ' {
                i += 1;
                start = true;
            } else if c == ' ' && start {
                return i;
            }
        }
        return i;
    }
}


